addons/binding/org.openhab.binding.co2signal/src/main/java/org/openhab/binding/co2signal/handler/CO2SignalHandler.java

```diff
@@ -0,0 +1,292 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.co2signal.handler;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.co2signal.CO2SignalBindingConstants;
+import org.openhab.binding.co2signal.internal.CO2SignalConfiguration;
+import org.openhab.binding.co2signal.internal.json.CO2SignalJsonResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link CO2SignalHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jens Viebig - Initial contribution
+ */
+public class CO2SignalHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(CO2SignalHandler.class);
+
+    private static final String URL = "https://api.co2signal.com/v1/latest?%QUERY%";
+
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+
+    private ScheduledFuture<?> refreshJob;
+
+    CO2SignalJsonResponse co2Response;
+
+    private Gson gson;
+
+    public CO2SignalHandler(@NonNull Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing CO2 Signal handler.");
+
+        CO2SignalConfiguration config = getConfigAs(CO2SignalConfiguration.class);
+        logger.debug("config apikey = (omitted from logging)");
+        logger.debug("config location = {}", config.location);
+        logger.debug("config countryCode = {}", config.countryCode);
+        logger.debug("config refresh = {}", config.refresh);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errorMsg = "Parameter 'apikey' is mandatory and must be configured";
```

Comment:
What if there is more than one error ;-)

Reply:
Fixed